File: src/mocha-tests/files/php/uncomment-feature/allLogTypes.php
<?php

function getUserById($id) {
  return ['id' => $id, 'name' => 'Test User', 'email' => 'test@example.com', 'active' => true];
}

function processUser($data) {
  return $data;
}

function processUserData($userData) {
  $userId = $userData['id'];
  // error_log("🚀 ~ userData: " . print_r($userData, true));
  
  if (!empty($userId)) {
    $user = getUserById($userId);
    // var_dump("🚀 ~ processUserData ~ user:", $user);
    
    if (empty($user['email'])) {
      // error_log("🚀 ~ processUserData ~ missing email: " . print_r($user, true));
    }
    
    if ($user['active']) {
      // error_log("🚀 ~ processUserData ~ processing user: " . print_r($user, true));
    }
    
    $result = [
      'user' => $user,
      'processed' => true
    ];
    
    // error_log("🚀 ~ processUserData ~ result: " . print_r($result, true));
    // print_r(["🚀 ~ processUserData ~ table data:" => $result]);
    
    return $result;
  }
  
  $processedUser = processUser($userData);
  // error_log("🚀 ~ processedUser: " . print_r($processedUser, true));
  
  // error_log("🚀 ~ multiline log: " . print_r($processedUser, true));
  
  return $processedUser;
}

File: src/mocha-tests/files/python/uncomment-feature/allLogTypes.py
import logging

def process_data(data):
    user_id = data['id']
    # print("🚀 ~ process_data ~ user_id:", user_id)

    if user_id:
        user = get_user(user_id)
        # print("🚀 ~ process_data ~ user:", user)

        if not user.get('email'):
            # logging.debug("🚀 ~ process_data ~ missing email: %s", user)

        result = {'user': user, 'processed': True}
        # logging.info("🚀 ~ process_data ~ result: %s", result)

        return result

    processed = process_user(data)
    # logging.warning("🚀 ~ processed: %s", processed)

    return processed
